src/routing/approved.rs

```diff
@@ -1534,26 +1534,27 @@ impl Approved {
                 commands.extend(self.send_sync(self.section.clone(), self.network.clone())?);
             }
 
+            let node_elder_change = if !old_is_elder && new_is_elder {
+                info!("Promoted to elder");
+                NodeElderChange::Promoted
+            } else if old_is_elder && !new_is_elder {
+                info!("Demoted");
+                self.section = self.section.trimmed(1);
+                self.network = Network::new();
+                self.section_keys_provider = SectionKeysProvider::new(None);
+                NodeElderChange::Demoted
+            } else {
+                NodeElderChange::None
+            };
+
             self.send_event(Event::EldersChanged {
                 prefix: *self.section.prefix(),
                 key: *self.section.chain().last_key(),
                 elders: self.section.elders_info().elders.keys().copied().collect(),
+                change: node_elder_change,
```

Comment:
the flag is only about self status change, may be name as `self_status_change` to make it more explicit ?
as the Event `EldersChanged` is actually about the whole section's elders.

Reply:
Yes, I agree the name was not good. I was wondering what the convention is now when referring to ourselves as node, is it `self` or `node`?

Reply:
I think both works. though I may prefer `self` more.